app/uk/gov/hmrc/membersprotectionsenhancements/controllers/actions/IdentifierAction.scala

```diff
@@ -43,37 +50,52 @@ class AuthenticatedIdentifierAction @Inject() (
   playBodyParsers: BodyParsers.Default
 )(implicit override val executionContext: ExecutionContext)
     extends IdentifierAction
-    with AuthorisedFunctions {
+    with AuthorisedFunctions
+    with Logging {
 
   override def invokeBlock[A](request: Request[A], block: IdentifierRequest[A] => Future[Result]): Future[Result] = {
 
-    implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromRequestAndSession(request, request.session)
+    implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromRequest(request)
 
     authorised(Enrolment(Constants.psaEnrolmentKey).or(Enrolment(Constants.pspEnrolmentKey)))
-      .retrieve(Retrievals.internalId.and(Retrievals.authorisedEnrolments)) {
+      .retrieve(internalId.and(affinityGroup).and(authorisedEnrolments)) {
 
-        case Some(internalId) ~ IsPSA(psaId) => block(AdministratorRequest(internalId, request, psaId.value))
-        case Some(internalId) ~ IsPSP(pspId) => block(PractitionerRequest(internalId, request, pspId.value))
-        case Some(_) ~ _ =>
+        case Some(internalId) ~ Some(affGroup) ~ IsPSA(psaId) =>
+          block(AdministratorRequest(affGroup, internalId, psaId.value, UserType.PSA, request))
+        case Some(internalId) ~ Some(affGroup) ~ IsPSP(pspId) =>
+          block(PractitionerRequest(affGroup, internalId, pspId.value, UserType.PSP, request))
+        case Some(_) ~ Some(_) ~ _ =>
           Future.failed(new UnauthorizedException("Not Authorised - Unable to retrieve credentials - externalId"))
         case _ =>
           Future.failed(new UnauthorizedException("Not Authorised - Unable to retrieve credentials - externalId"))
       }
-      .recoverWith { case _ =>
-        Future.successful(BadRequest(""))
+      .recoverWith {
+        case e: MissingBearerToken =>
+          logger.warn(
+            s"[AuthenticatedIdentifierAction][invokeBlock - MissingBearerToken] An unexpected error occurred: ${e.printStackTrace()}"
+          )
+          Future.successful(Unauthorized(Json.toJson(InvalidBearerTokenError)))
+        case e: AuthorisationException =>
+          logger.warn(
+            s"[AuthenticatedIdentifierAction][invokeBlock - AuthorisationException] An unexpected error occurred: ${e.printStackTrace()}"
+          )
+          Future.successful(Unauthorized(Json.toJson(UnauthorisedError)))
+        case error =>
```

Comment:
The issue with putting a wildcard case here is that it usually means that if an exception is thrown by any controller that uses auth it will show up as an auth error because the exception gets caught here rather than going to the error handler. I.e if the connector throws an unhandled exception we will see this case here being hit

Reply:
We have a ticket to handle all error handling mechanism. We will discuss all the scenarios and error response to the FE and implement them accordinly.